std/datetime/date.d

```diff
@@ -2869,7 +2869,13 @@ public:
     {
         import std.format : format;
         try
-            return format("%sT%s", _date.toISOString(), _tod.toISOString());
```

Comment:
I've thought about it yes. Before I did anything with that I wanted to put forth a RFC PR about https://issues.dlang.org/show_bug.cgi?id=17420

Regardless that change wouldn't be part of this PR.